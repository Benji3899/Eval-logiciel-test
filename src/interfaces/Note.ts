// src/interfaces/Note.ts
export interface Note {
    id: string;
    title: string;
    grade:number;
    // score: number;
    comment: string;
    // createdAt: string;
    date: string;
}
