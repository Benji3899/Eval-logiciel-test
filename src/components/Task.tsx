import { Task as TaskType } from '../interfaces/Task';

import styles from './Task.module.css';

interface TaskProps {
  task: TaskType;
}

export function Task({ task }: TaskProps) {
  return <li className={styles.wrapper}>{task.title}</li>;
}
