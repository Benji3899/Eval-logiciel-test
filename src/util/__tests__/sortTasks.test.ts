import { test, expect } from 'vitest';
import { sortTasks } from '../sortTasks';
import { Task } from '../../interfaces/Task';

const tasks: Task[] = [
  {
    id: '1',
    title: 'Task 1',
    created_at: new Date('2023-01-01T00:00:00').getTime(),
    done_at: undefined,
  },
  {
    id: '2',
    title: 'Task 2',
    created_at: new Date('2023-01-02T00:00:00').getTime(),
    done_at: new Date('2023-01-04T00:00:00').getTime(),
  },
  {
    id: '3',
    title: 'Task 3',
    created_at: new Date('2023-01-02T00:00:00').getTime(),
    done_at: undefined,
  },
];

test('sorts tasks correctly by done at', () => {
  const result = sortTasks(tasks);

  expect(result).toMatchSnapshot();
});

test('Tasks should be sorted correctly when there is no done from the past to the future', () => {
  const tasks: Task[] = [
    {
      id: '1',
      title: 'Task 1',
      created_at: new Date('1995-12-17T00:00:00').getTime(),
      done_at: undefined,
    },
    {
      id: '2',
      title: 'Task 2',
      created_at: new Date('2023-01-01T00:00:00').getTime(),
      done_at: undefined,
    },
    {
      id: '3',
      title: 'Task 3',
      created_at: new Date('2024-01-01T00:00:00').getTime(),
      done_at: undefined,
    },
  ];
  const result = sortTasks(tasks);

  expect(result).toMatchSnapshot();
});

test('Completed task should order at the botton from the most recent to the most old', () => {
  const tasks: Task[] = [
    {
      id: '1',
      title: 'Task 1',
      created_at: new Date('1995-12-17T00:00:00').getTime(),
      done_at: undefined,
    },
    {
      id: '4',
      title: 'Task 4',
      created_at: new Date('2024-01-01T00:00:00').getTime(),
      done_at: new Date('2024-01-02T00:00:00').getTime(),
    },
    {
      id: '2',
      title: 'Task 2',
      created_at: new Date('2023-01-01T00:00:00').getTime(),
      done_at: undefined,
    },
    {
      id: '3',
      title: 'Task 3',
      created_at: new Date('2024-01-01T00:00:00').getTime(),
      done_at: undefined,
    },

    {
      id: '5',
      title: 'Task 5',
      created_at: new Date('2023-01-01T00:00:00').getTime(),
      done_at: new Date('2023-01-02T00:00:00').getTime(),
    },
  ];
  const result = sortTasks(tasks);
  expect(result).toMatchSnapshot();
});
