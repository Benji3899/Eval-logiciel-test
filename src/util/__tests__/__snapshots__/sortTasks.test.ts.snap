// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Completed task should order at the botton from the most recent to the most old 1`] = `
[
  {
    "created_at": 819154800000,
    "done_at": undefined,
    "id": "1",
    "title": "Task 1",
  },
  {
    "created_at": 1672527600000,
    "done_at": undefined,
    "id": "2",
    "title": "Task 2",
  },
  {
    "created_at": 1704063600000,
    "done_at": undefined,
    "id": "3",
    "title": "Task 3",
  },
  {
    "created_at": 1704063600000,
    "done_at": 1704150000000,
    "id": "4",
    "title": "Task 4",
  },
  {
    "created_at": 1672527600000,
    "done_at": 1672614000000,
    "id": "5",
    "title": "Task 5",
  },
]
`;

exports[`Tasks should be sorted correctly when there is no done from the past to the future 1`] = `
[
  {
    "created_at": 819154800000,
    "done_at": undefined,
    "id": "1",
    "title": "Task 1",
  },
  {
    "created_at": 1672527600000,
    "done_at": undefined,
    "id": "2",
    "title": "Task 2",
  },
  {
    "created_at": 1704063600000,
    "done_at": undefined,
    "id": "3",
    "title": "Task 3",
  },
]
`;

exports[`sorts tasks correctly by done at 1`] = `
[
  {
    "created_at": 1672527600000,
    "done_at": undefined,
    "id": "1",
    "title": "Task 1",
  },
  {
    "created_at": 1672614000000,
    "done_at": undefined,
    "id": "3",
    "title": "Task 3",
  },
  {
    "created_at": 1672614000000,
    "done_at": 1672786800000,
    "id": "2",
    "title": "Task 2",
  },
]
`;
